src/components/PopupWithForm.js

```diff
@@ -4,6 +4,16 @@ export default class PopupWithForm extends Popup {
     super({ popup });
     this._popupForm = this._popup.querySelector(".form");
     this._formSubmit = formSubmit;
+    this._submitButton = this._popupForm.querySelector(".form__button");
+    this._submitButtonText = this._submitButton.textContent;
+  }
+
+  renderLoading(isLoading, restoredText) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = restoredText;
```

Comment:
`restoredText` is not needed because you have `this._submitButtonText` from line 8


```js

      this._submitButton.textContent = this._submitButtonText;

```